src/Client/Core/Entites/DurableEntityClient.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.DurableTask.Entities;
+
+namespace Microsoft.DurableTask.Client.Entites;
+
+/// <summary>
+/// A client for interacting with entities.
+/// </summary>
+public abstract class DurableEntityClient
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DurableEntityClient"/> class.
+    /// </summary>
+    /// <param name="name">The name of the client.</param>
+    protected DurableEntityClient(string name)
+    {
+        this.Name = name;
+    }
+
+    /// <summary>
+    /// Gets the name of the client.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// Signals an entity to perform an operation.
+    /// </summary>
+    /// <param name="id">The ID of the entity to signal.</param>
+    /// <param name="operationName">The name of the operation.</param>
+    /// <param name="input">The input for the operation.</param>
+    /// <param name="options">The options to signal the entity with.</param>
+    /// <param name="cancellation">The cancellation token to cancel enqueuing of the operation.</param>
+    /// <returns>A task that completes when the message has been reliably enqueued.</returns>
+    /// <remarks>This does not wait for the operation to be processed by the receiving entity.</remarks>
+    public abstract Task SignalEntityAsync(
+        EntityInstanceId id,
+        string operationName,
+        object? input = null,
+        SignalEntityOptions? options = null,
+        CancellationToken cancellation = default);
+
+    /// <summary>
+    /// Signals an entity to perform an operation.
+    /// </summary>
+    /// <param name="id">The ID of the entity to signal.</param>
+    /// <param name="operationName">The name of the operation.</param>
+    /// <param name="options">The options to signal the entity with.</param>
+    /// <param name="cancellation">The cancellation token to cancel enqueuing of the operation.</param>
+    /// <returns>A task that completes when the message has been reliably enqueued.</returns>
+    /// <remarks>This does not wait for the operation to be processed by the receiving entity.</remarks>
+    public virtual Task SignalEntityAsync(
```

Comment:
I am a bit concerned about this overload (not passing an input) creating a bad pitfall where users forget to pass an input to an operation that must have one specified (i.e. does not have a default). The problem is that they won't see an exception in the client because signals don't wait for a response. The exception will be quietly buried in some log somewhere.

Reply:
IMO this pitfall may be worse than the utility provided by default arguments. I don't actually recall ever having a need for default arguments on entity operations so we should weigh the extra complexity and error modes of supporting this feature.

Reply:
> creating a bad pitfall where users forget to pass an input to an operation that must have one specified

What alternative do you suggest? Users explicitly pass `null` for the input? I don't see this as any _new_ pitfall. Users can mistakenly pass `null` for input, or ignore it with optional params. Additionally, a similar pitfall already exists with passing in incompatible input for an operation.

To me, users already need to be aware of what input is needed - if any - for a given operation. So, I don't see this as a new pitfall - it already exists.

Reply:
> I don't see this as any new pitfall. 

true, the current implementation seems to have the same problem. And I can't really think of a better solution (short of using interfaces and proxies which is out of scope).

> The reason I include this overload is to avoid the accidental cancellation token as input - which IMO is an even worse pitfall. 

Thanks for pointing that out, I definitely did not see that. Makes sense.